ethpm/package.py

```diff
@@ -11,27 +12,69 @@
 
 
 class Package(object):
-    def __init__(self, lockfile):
+    web3 = None
+
+    def __init__(self, package_identifier, web3=None):
         """
         A lockfile can be:
         - filesystem path
         - parsed lockfile JSON
         - lockfile URI
         """
-        if not os.path.exists(lockfile):
+        self.web3 = web3
+
+        if not os.path.exists(package_identifier):
             raise ValidationError
 
-        self.package_identifier = lockfile
+        self.package_identifier = package_identifier
 
-        schema_data = json.load(open('./ethpm/lockfileSpecification.json'))
-        package_data = json.load(open(lockfile))
+        schema_data = self._load_json('./ethpm/assets/release-lockfile.schema.v1.json')
+        package_data = self._load_json(package_identifier)
 
         try:
             validate(package_data, schema_data)
         except jsonValidationError:
             raise ValidationError
         self.package_data = package_data
 
+    def _load_json(self, path):
+        return json.load(open(path))
+
+    def set_default_web3(self, web3):
+        """
+        Set the default Web3 instance.
+        """
+        self.web3 = web3
+
+    def get_contract_type(self, name, web3=None):
+        """
+        API to generate a contract factory class.
+        """
+        self._validate_contract_name(name)
+        current_web3 = None
+
+        if web3 is not None:
+            current_web3 = web3
+        else:
+            current_web3 = self.web3
+        if current_web3 is None:
+            raise ValidationError
```

Comment:
This should get a message added so that when it happens it's clear what went wrong.